cljfmt/src/cljfmt/config.clj

```diff
@@ -59,6 +59,46 @@
 (defn- directory? [path]
   (some-> path io/file .getAbsoluteFile .isDirectory))
 
+(defn- -load-config
+  [path]
+  (->> (some-> path read-config)
+       (merge default-config)
+       (convert-legacy-keys)))
+
+(defn- deep-merge
```

Comment:
just stole quick from medley as an example

Reply:
Merging configurations isn't a deep-merge normally in cljfmt, so it shouldn't be for `:extra-configs`. Instead, it should be a normal merge.

Reply:
Well, I don't really think it would make sense personally for each config to be wiping each others :extra-indents, as the whole point in my mind would be for them to be merged together. I assume you mean that it would make sense to just `merge` the :extra-indents into one another? I can do that pretty easily if that works?

Reply:
I've updated it to merge the :extra-indents maps together, and then merge the final :extra-indents back into the base config. If that's not what you meant or Ok let me know

Reply:
Hm, that's a fair point. Does `deps.edn` solve this in any particular way?

I don't like the idea of recursively merging configs, as that makes the process purely additive, and makes it different from how the default configuration is handled, and how options are handled.

I'm beginning to think that this issue might be better solved via a different means entirely. For example, if the use-case is just supporting extra indentations in libraries, we could create a indent resource in a standard location.

Reply:
> Hm, that's a fair point. Does deps.edn solve this in any particular way?

I'm not familiar with deps.edn at all really, so unsure

> I don't like the idea of recursively merging configs, as that makes the process purely additive, and makes it different from how the default configuration is handled, and how options are handled.

In my mind, this is new behavior is somewhat consistent with how :extra-indents works today, adding to the default :indents . I was also thinking that, since the base config has the final say, it could overwrite any values pulled in from downstream that it doesn't agree with, but that may be less than ideal, not sure.

> I'm beginning to think that this issue might be better solved via a different means entirely. For example, if the use-case is just supporting extra indentations in libraries, we could create a indent resource in a standard location.

I'm not sure what you mean by this, but I'm open to other ideas/solutions

Reply:
> I'm not sure what you mean by this, but I'm open to other ideas/solutions

clj-kondo has a mechanism for [importing configurations from libraries](https://github.com/clj-kondo/clj-kondo/blob/master/doc/config.md#exporting-and-importing-configuration). We could do something similar with cljfmt.

The way it would work is that cljfmt can optionally be given a classpath, which it will then search through to find indentation files in some predictable location. For example `cljfmt.exports/*/*/indent.edn`.